src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -3217,12 +3210,12 @@ private static bool IsNegativeConstantForArraySize(BoundExpression expression)
             MethodSymbol constructor,
             DiagnosticBag diagnostics)
         {
-            Binder initializerBinder = initializerArgumentListOpt == null ? this : this.GetBinder(initializerArgumentListOpt);
+            Binder initializerBinder = initializerArgumentListOpt == null ? this : (this.GetBinder(initializerArgumentListOpt) ?? this);
             Debug.Assert(initializerBinder != null);
 
             var result = initializerBinder.BindConstructorInitializerCore(initializerArgumentListOpt, constructor, diagnostics);
 
-            if (initializerArgumentListOpt != null)
+            if (initializerArgumentListOpt != null && this != initializerBinder)
```

Comment:
>this != initializerBinder [](start = 54, length = 25)

Can you please explain this test `(this != initializerBinder)? What kind of source code requires this test?